src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -692,6 +699,53 @@ def setUp(self):
                 manager_class_name, None, manager_subprocess, manager_mpi_type)
             self.set_job_manager_timeout()
 
+    def write_string_to_logfile(self, s):
+        """Write a string to the server log
+
+        Args:
+            s (string): string to write to log file.  Log message will be in
+                the following format:
+                  $date $hostname DAOS[$pid/0/6] rpc EMIT \
+                    src/cart/crt_rpc.c:53 crt_hdlr_ctl_log_add_msg() $s
+
+        """
+
+        # Fetch attachinfo data from server
+        self.agent_managers[0].dump_attachinfo()
+        self.log.info("Agent attachinfo: %s",
+                      self.agent_managers[0].attachinfo)
+
+        a = self.agent_managers[0].attachinfo
+
+        # Filter log messages from attachinfo content
+        l = [x for x in a if re.match(r"^(name\s|size\s|all|\d+\s)", x)]
+        attach_info_contents = "\n".join(l)
+
+        # Write an attach_info_tmp file in this directory for cart_ctl to use
+        tmpdir = tempfile.gettempdir()
+        attachinfo_file_name = tmpdir + "/daos_server.attach_info_tmp"
+
+        file1 = None
+        try:
+            file1 = open(attachinfo_file_name, 'w')
+            file1.write(attach_info_contents)
+        finally:
+            file1.close()
+
+        cp_command = "sudo cp {} {}".format(attachinfo_file_name, ".")
+        run_command(cp_command, verbose=True, raise_exception=False)
+
+        # Compose and run cart_ctl command
+        cart_ctl = CartCtl()
+        cart_ctl.add_log_msg.value = "add_log_msg"
+        cart_ctl.rank.value = 0
+        cart_ctl.cfg_path.value = "."
+        cart_ctl.group_name.value = "daos_server"
+        cart_ctl.m.value = s
```

Comment:
this shows that we are sending cart_ctl command to rank=0; we should be sending to all server ranks instead. cart_ctl has  "--rank all" option

Reply:
https://github.com/daos-stack/daos/pull/5384/commits/a00d55085bf3aea54daad85f937a8a3113b741d0